Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -233,10 +233,14 @@ void SantaDecisionManager::AddToCache(
     case ACTION_REQUEST_BINARY:
       decision_cache->set(identifier, val, 0);
       break;
+    case ACTION_RESPOND_ACK:
+      decision_cache->set(identifier, val, ((uint64_t)ACTION_REQUEST_BINARY << 56));
+      break;
     case ACTION_RESPOND_ALLOW:
     case ACTION_RESPOND_DENY:
-      decision_cache->set(
-          identifier, val, ((uint64_t)ACTION_REQUEST_BINARY << 56));
+      if (decision_cache->set(identifier, val, ((uint64_t)ACTION_REQUEST_BINARY << 56))) {
```

Comment:
Can you add a TODO above this line to make calling set twice unnecessary - having both calls will involve finding and locking the bucket (or buckets) multiple times.

Reply:
Done.